src/du_parameters/frequency.py

```diff
@@ -0,0 +1,526 @@
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Define Frequency, GSCN and ARFCN classes by overloading arithmetic operators.
+
+Handle arithmetic operations with different types of objects.
+"""
+
+import logging
+from dataclasses import dataclass
+from decimal import Decimal
+
+logger = logging.getLogger(__name__)
+
+CONFIG_CONSTANT_THREE = Decimal("3")
+CONFIG_CONSTANT_TWO = Decimal("2")
+
+
+class GetRangeFromFrequencyError(Exception):
+    """Exception raised when frequency is not appropriate for any frequency range."""
+
+    pass
+
+
+class GetRangeFromGSCNError(Exception):
+    """Exception raised when GSCN is not appropriate for any frequency range."""
+
+    pass
+
+
+class Frequency(Decimal):
+    """Represents a frequency with unit conversion methods and arithmetic capabilities."""
+
+    class Units:
+        """Define Units for Frequency class."""
+
+        KHZ = 1000
+        MHZ = 1000 * 1000
+
+    @classmethod
+    def from_khz(cls, value: int | str) -> "Frequency":
+        """Convert KHz to Frequency."""
+        return cls(Decimal(value) * cls.Units.KHZ)
+
+    @classmethod
+    def from_mhz(cls, value: int | str) -> "Frequency":
+        """Convert MHz to Frequency."""
+        return cls(Decimal(value) * cls.Units.MHZ)
+
+    def __add__(self, other: "Frequency | Decimal | str | int") -> "Frequency":
+        """Add Frequency to others.
+
+        Args:
+            other (Frequency | Decimal | str | int): The value to add.
+
+        Returns:
+            Frequency: The resulting Frequency after the addition.
+        """
+        if not isinstance(other, (Frequency, Decimal, int, str)):
+            raise NotImplementedError(f"Unsupported type for addition: {type(other)}")
+        return Frequency(super().__add__(Decimal(other)))
```

Comment:
I think we could simplify this by trying to convert `other` to a `Decimal`:

```suggestion
        try:
            return Frequency(super().__add__(Decimal(other)))
        except ValueError:
            raise NotImplementedError
```

We then do not have to list all the supported types.